cloud/edgecontroller/pkg/controller/manager/location.go

```diff
@@ -56,6 +63,17 @@ func (lc *LocationCache) newNodes(oldNodes []string, node string) []string {
 func (lc *LocationCache) AddOrUpdatePod(pod v1.Pod) {
 	configMaps, secrets := lc.PodConfigMapsAndSecrets(pod)
 
+	//update nodeRole map
+	lc.UpdateNodeRoleMap(meta_v1.ListOptions{})
```

Comment:
On each update in any Pod, we update the list of all nodes. Can we add a parameter "node" in UpdateNodeRoleMap function so that we only update the status of the node on which the pod is scheduled.